# HG changeset patch
# User Markus Bergholz <markuman+octave@gmail.com>
# Date 1412709077 -7200
#      Tue Oct 07 21:11:17 2014 +0200
# Node ID e1713e281ac540f4cee71be966a59e2decb795d0
# Parent  a834b3151c7e3e381310f637045c4e52233ae7c2
rat.m: Remove unnecessary blank line at start of output (bug #43374).

* rat.m: Check whether numerator is null before expanding it to the
length of s.  Add BIST test for correct behavior.

diff --git a/scripts/general/rat.m b/scripts/general/rat.m
--- a/scripts/general/rat.m
+++ b/scripts/general/rat.m
@@ -137,26 +137,29 @@ function [n,d] = rat (x,tol)
           s = [s(1:end-1), " + 1/(", int2str(step), ")"];
         endif
       endwhile
       s = [s, repmat(")", 1, j-2)];
       n_nc = columns (n);
       s_nc = columns (s);
       if (n_nc > s_nc)
         s(:,s_nc+1:n_nc) = " ";
-      elseif (s_nc > n_nc)
+      elseif (s_nc > n_nc && n_nc != 0)
         n(:,n_nc+1:s_nc) = " ";
       endif
       n = cat (1, n, s);
     endfor
   endif
 
 endfunction
 
 
 %!test
 %! [n, d] = rat ([0.5, 0.3, 1/3]);
 %! assert (n, [1, 3, 1]);
 %! assert (d, [2, 10, 3]);
 
+## bug #43374
+%!assert (eval (rat (0.75)), [0.75])
+
 %!error rat ();
 %!error rat (1, 2, 3);
 
